tests/testapp/tests/test_syncsession_methods.py

```diff
@@ -0,0 +1,251 @@
+                                '{}:user:interaction'.format(self.data['user3'].id)]
+        self.data['sc']._queue_into_buffer(self.filter_prefixes, self.fsics)
+        buffer_ids = Buffer.objects.values_list('model_uuid', flat=True)
+        rmcb_ids = RecordMaxCounterBuffer.objects.values_list('model_uuid', flat=True)
+        self.assertNotIn(self.data['user2'].id, buffer_ids)
+        self.assertNotIn(self.data['user2'].id, rmcb_ids)
+        for i in self.data['user3_sumlogs']:
+            self.assertIn(i.id, buffer_ids)
+            self.assertIn(i.id, rmcb_ids)
+        for i in self.data['user3_interlogs']:
+            self.assertIn(i.id, buffer_ids)
+            self.assertIn(i.id, rmcb_ids)
+
+    def test_partition_prefix_buffering(self):
+        self.filter_prefixes = ['{}'.format(self.data['user2'].id)]
+        self.data['sc']._queue_into_buffer(self.filter_prefixes, self.fsics)
+        buffer_ids = Buffer.objects.values_list('model_uuid', flat=True)
+        rmcb_ids = RecordMaxCounterBuffer.objects.values_list('model_uuid', flat=True)
+        self.assertIn(self.data['user2'].id, buffer_ids)
+        self.assertIn(self.data['user2'].id, rmcb_ids)
+        for i in self.data['user2_sumlogs']:
+            self.assertIn(i.id, buffer_ids)
+            self.assertIn(i.id, rmcb_ids)
+        for i in self.data['user2_interlogs']:
+            self.assertIn(i.id, buffer_ids)
+            self.assertIn(i.id, rmcb_ids)
+        self.assertNotIn(self.data['user3'].id, buffer_ids)
+        self.assertNotIn(self.data['user3'].id, rmcb_ids)
+
+    def test_partition_and_fsic_buffering(self):
+        self.filter_prefixes = ['{}:user:summary'.format(self.data['user1'].id)]
+        self.fsics = {self.data['group1_id']: 1}
+        self.data['sc']._queue_into_buffer(self.filter_prefixes, self.fsics)
+        buffer_ids = Buffer.objects.values_list('model_uuid', flat=True)
+        rmcb_ids = RecordMaxCounterBuffer.objects.values_list('model_uuid', flat=True)
+        for i in self.data['user1_sumlogs']:
+            self.assertIn(i.id, buffer_ids)
+            self.assertIn(i.id, rmcb_ids)
+        for i in self.data['user2_sumlogs']:
+            self.assertNotIn(i.id, buffer_ids)
+            self.assertNotIn(i.id, rmcb_ids)
+        for i in self.data['user3_sumlogs']:
+            self.assertNotIn(i.id, buffer_ids)
+            self.assertNotIn(i.id, rmcb_ids)
+
+
+class BufferIntoStoreTestCase(TestCase):
+
+    def setUp(self):
+        self.data = {}
+        DatabaseIDModel.objects.create()
+        (self.current_id, _) = InstanceIDModel.get_or_create_current_instance()
+
+        # create controllers for app/store/buffer operations
+        self.data['mc'] = MorangoProfileController('facilitydata')
+        self.data['sc'] = SyncClient('host', 'facilitydata')
+
+        self.data.update(create_buffer_and_store_dummy_data(self.data['sc'].transfer_session_id))
```

Comment:
Well since I set it at instantiation of the `SyncClient`, but will need to update tests once we change that.